dotcom-rendering/src/web/components/EnhancePinnedPost.importable.tsx

```diff
@@ -91,40 +94,26 @@ export const EnhancePinnedPost = () => {
 	// and emit ophan events when the pinned post goes out of view
 	useEffect(() => {
 		if (!pinnedPost) return;
-
 		const pinnedPostTiming = initPerf('pinned-post-view-duration');
```

Comment:
I think its best to avoid refs

Reply:
> [as per the React docs](https://reactjs.org/docs/refs-and-the-dom.html).

Oh those docs are unfortunately out of date and mainly refer to pre-hooks refs and their use as handles to dom elements. The React docs are in dire need of an update!

A more up to date reference is here:
https://reactjs.org/docs/hooks-reference.html#useref

Also the (vastly improved) beta of the new React docs on refs are here:
https://beta.reactjs.org/apis/useref

Reply:
Thank you both for your time and reviews. 
We had a session with @arelra about `useRef` usage in react which I briefly explain here:

In react docs [this article](https://reactjs.org/docs/refs-and-the-dom.html) is discouraging the usage of `useRef` but the article seems to be an explanation of useRef only for accessing DOM elements within the class based components. 

In the other 2 docs https://reactjs.org/docs/hooks-reference.html#useref & https://beta.reactjs.org/apis/useref it's explaining the `useRef` as a hook which is how we're doing in our functional components. 

Some interesting points from these docs are: 
- useRef is a React Hook that lets you reference a value that’s not needed for rendering
- By using a ref, you ensure that:
    1. You can store information between re-renders
    2. Changing it does not trigger a re-render

In our case, pinnedPostTiming is not needed for rendering. And we just want to store the object between re-renders. 
If we do it this way: `const pinnedPostTiming = initPerf('pinned-post-view-duration');` , initPerf is called on every re-render. I added a console log within `initPerf` and captured the screen video that shows every re-render is also logging **initPerf is instansiated**:
```
	if (name === 'pinned-post-view-duration') {
		console.log('initPerf is instansiated');
	}
``` 


https://user-images.githubusercontent.com/15894063/164046076-6294b9fc-a1ca-4181-9516-603094bf3dd3.mov

I then used it within a `useRef` as you see below:
```
	const pinnedPostTiming =
		useRef<{ start: () => void; end: () => number; clear: () => void }>();

	useEffect(() => {
		pinnedPostTiming.current = initPerf('pinned-post-view-duration');
	}, []);
``` 

Again I captured the screen video, as you can see below, there's only one call to initPerf this time: 

https://user-images.githubusercontent.com/15894063/164046848-76afd754-1ad4-4c3c-a3bf-590873524e22.mov

So based on this exploration I think it'd be probably better to use `useRef` for `pinnedPostTiming` to avoid redundant calls to `initPerf` but  I'm more than happy to hear any views about this :)

Reply:
This is an excellent explanation. Keep using `useRef`!

Reply:
Perfectly explained @marjisound . Great work!